webots_ros2_examples/README.md

```diff
@@ -0,0 +1,139 @@
+```
+
+#### Troubleshooting
+
+If you see import failures or some indications saying that `WEBOTS_HOME` is incorrectly set, make sure your environment variables are [configured for extern controllers](https://www.cyberbotics.com/doc/guide/running-extern-robot-controllers?version=master#environment-variables).
+
+
+## TurtleBot3 Burger
+
+![TurtleBot3 Burger in Webots](./assets/turtlebot3_burger.png)
+
+To run TurtleBot3 Burger simulation you can use the universal launcher:
+```bash
+ros2 launch webots_ros2_core robot_launch.py \
+    executable:=webots_differential_drive_node \
+    node_parameters:=$(ros2 pkg prefix webots_ros2_examples --share)/resource/turtlebot3_burger.yaml \
+    world:=$(ros2 pkg prefix webots_ros2_examples --share)/worlds/turtlebot3_burger_example.wbt
+```
+
+If you have [`turtlebot3`](https://github.com/ROBOTIS-GIT/turtlebot3) package installed you can start `cartographer` as:
+```bash
+ros2 launch turtlebot3_cartographer cartographer.launch.py \
+    use_sim_time:=true
+```
+
+Also, you can start `navigation2` using a custom map:
+```bash
+ros2 launch turtlebot3_navigation2 navigation2.launch.py \
+    use_sim_time:=true \
+    map:=$(ros2 pkg prefix webots_ros2_examples --share)/resource/turtlebot3_burger_example_map.yaml
+```
+
+Make sure you set initial pose by clicking `2D Pose Estimate` button in RViz or by executing the following command:
+```bash
+ros2 topic pub --once /initialpose geometry_msgs/msg/PoseWithCovarianceStamped '{
+    "header": { "frame_id": "map" },
+    "pose": {
+        "pose": {
+            "position": { "x": 0.0, "y": 0.0, "z": 0.0 },
+            "orientation": { "x": 0.0, "y": 0.0, "z": 0.0, "w": 1.0 }
+        }
+    }
+}'
+```
+
+> On the [official TurtleBot3 website](https://emanual.robotis.com/docs/en/platform/turtlebot3/overview/#turtlebot3) you can find more information about [navigation](https://emanual.robotis.com/docs/en/platform/turtlebot3/ros2_navigation2/) and [SLAM](https://emanual.robotis.com/docs/en/platform/turtlebot3/ros2_slam/) configuration.
+
+This will expose necessary topics:
+```bash
+$ ros2 topic list -t
+/scan [sensor_msgs/msg/LaserScan]
+/cmd_vel [geometry_msgs/msg/Twist]
+/imu [sensor_msgs/msg/Imu]
+/joint_states [sensor_msgs/msg/JointState]
+/odom [nav_msgs/msg/Odometry]
+/robot_description [std_msgs/msg/String]
+/tf [tf2_msgs/msg/TFMessage]
+/tf_static [tf2_msgs/msg/TFMessage]
+```
+
+## Khepera IV
+
+![Khepera IV in Webots](./assets/khepera4.png)
+
+Similarly to TurtleBot3 Burger you can start Khepera IV robot as well:
+```bash
+ros2 launch webots_ros2_core robot_launch.py \
+    executable:=webots_differential_drive_node \
+    node_parameters:=$(ros2 pkg prefix webots_ros2_examples --share)/resource/khepera4.yaml \
+    world:=$(ros2 pkg prefix webots_ros2_examples --share)/worlds/khepera4_example.wbt
+```
+
+For this robot you should expect the following topics:
+```bash
+$ ros2 topic list -t
+/camera/camera_info [sensor_msgs/msg/CameraInfo]
```

Comment:
I just found that echoing this topic does not print anything:
```shell
ros2 topic echo /camera/camera_info

``

Reply:
It is latched topic, it should work if you specify QoS:
```
ros2 topic echo --qos-profile services_default --qos-durability transient_local /camera/camera_info
```